workbench-core/example/infrastructure/integration-tests/tests/datasets/accessPermissions.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { dataSetPrefix } from '@aws/workbench-core-example-app/lib/configs/constants';
+import { CreateUser } from '@aws/workbench-core-user-management';
+import { v4 as uuidv4 } from 'uuid';
+import ClientSession from '../../support/clientSession';
+import Dataset from '../../support/resources/datasets/dataset';
+import Setup from '../../support/setup';
+import HttpError from '../../support/utils/HttpError';
+
+describe('DataSets access permissions integration tests', () => {
+  const setup: Setup = new Setup();
+  const mockBadValue: string = 'fake-data';
+  let adminSession: ClientSession;
+  let user: CreateUser;
+
+  beforeEach(() => {
+    expect.hasAssertions();
+  });
+
+  beforeAll(async () => {
+    adminSession = await setup.getDefaultAdminSession();
+    user = {
+      firstName: 'Test',
+      lastName: 'User',
+      email: `success+create-user-${uuidv4()}@simulator.amazonses.com`
+    };
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('addDataSetAccessPermissions', () => {
+    it('Adds a read-only access permission.', async () => {
+      let response = await adminSession.resources.datasets.create({}, true);
+      const dataSetId: string = response.data.id;
+      response = await adminSession.resources.groups.create({}, true);
+      const { groupId } = response.data;
+      await expect(
+        (adminSession.resources.datasets.children.get(dataSetId) as Dataset).addAccess({
+          permission: {
+            identityType: 'GROUP',
+            identity: groupId,
+            accessLevel: 'read-only'
```

Comment:
i see a lot of copy and pasted strings: do we have constants for the domains specific strings for AuthZ like `GROUP`, `read-only`, `read-write`, etc?

Reply:
I will look into this. I can use constants if we define them in `base` or `datasets`. I prefer not to use them if they are defined in `authz` as that will encourage making the package coupling stronger. I would eventually like to refactor our code to decouple more.